Cod3rsGrowth.Web/wwwroot/webapp/controller/CriarRaca.controller.js

```diff
@@ -31,28 +49,101 @@ sap.ui.define(
+                this._limparInputs();
+                const tempoParaVisualizarMensagem = 2000;
+                setTimeout(() => this.onNavBack(), tempoParaVisualizarMensagem);
+              } catch (erros) {
+                this._exibirErros(erros);
+              }
+            }
+          } else {
+            if (this._validarNovaRaca(this.raca)) {
+              try {
+                const racaEditada = await RacaService.editarRaca(this.raca);
+                const mensagemDeSucesso = "Raça editada com sucesso!";
+                const tituloDaMessageBox = "Sucesso";
+                sucessoMessageBox(mensagemDeSucesso, tituloDaMessageBox);
+                this._limparInputs();
+                const tempoParaVisualizarMensagem = 2000;
+                setTimeout(() => this.onNavBack(), tempoParaVisualizarMensagem);
+              } catch (erros) {
+                this._exibirErros(erros);
+              }
+            }
+          }
+        },
 
-          if (this._validarNovaRaca(this.raca)) {
+        onNavToListaDeRacas: function () {
+          const rotaListaDeRacas = "listaDeRacas";
+          this.onNavTo(rotaListaDeRacas);
+        },
+
+        _carregarDadosDaRacaSelecionada: async function (oEvent) {
+          if (oEvent.getParameter("arguments").id) {
             try {
-              const racaCriada = await RacaService.adicionarRaca(this.raca);
-              const mensagemDeSucesso = "Raça adicionada com sucesso!";
-              MessageBox.show(mensagemDeSucesso, {
-                icon: sap.m.MessageBox.Icon.SUCCESS,
-                title: "Sucesso",
-                dependentOn: this.getView(),
-              });
-              this._limparInputs();
-              const tempoParaVisualizarMensagem = 2000;
-              setTimeout(() => this.onNavBack(), tempoParaVisualizarMensagem);
+              const idRaca = oEvent.getParameter("arguments").id;
+              const raca = await RacaService.obterRaca(idRaca);
+              const modelo = new JSONModel(raca);
+              const modeloRaca = "raca";
+
+              this.getView().setModel(modelo, modeloRaca);
             } catch (erros) {
               this._exibirErros(erros);
             }
           }
         },
 
-        onNavToListaDeRacas: function () {
-          const rotaListaDeRacas = "listaDeRacas";
-          this.onNavTo(rotaListaDeRacas);
+        _atualizarTextoDaPáginaCriar: function () {
+          this._atualizarTituloDaPaginaCriar();
+          this._atualizarLabelDoBotaoAdicionar();
+        },
+
+        _atualizarTextoDaPáginaEditar: function () {
+          this._atualizarTituloDaPaginaEditar();
+          this._atualizarLabelDoBotaoEditar();
+        },
+
+        _atualizarTituloDaPaginaCriar: function () {
+          const idPaginaCriacao = "paginaCriarRaca";
+          const oBundle = this.getView().getModel("i18n").getResourceBundle();
+          const novoTitulo = oBundle.getText("TituloPaginaCriarRaca");
```

Comment:
const oBundle = this.getView().getModel("i18n").getResourceBundle();
const novoTitulo = oBundle.getText("TituloPaginaCriarRaca");

isso aqui coloca numa funcao na baseController, é uma coisa que acaba utilizando bastante. uma funcao de nome obterMensagemI18n(chave) seria o ideal